src/shared/hooks/useTable.ts

```diff
@@ -28,15 +28,24 @@ export const useTable = (
 
   const { ownerId } = workspaces.useData() || {};
 
-  const params = {
-    ownerId,
-    limit,
-    search: searchValue,
-    page,
-    ...(ordering && { ordering }),
-  };
+  const params = useMemo(
+    () => ({
+      ownerId,
+      limit,
+      search: searchValue,
+      page,
+      ...(ordering && { ordering }),
+    }),
+    [ownerId, limit, searchValue, page, ordering],
+  );
+
+  // Ref required to eliminate closure reference errors and reduce redundant rerenders
+  const paramsRef = useRef(params);
+  useEffect(() => {
+    paramsRef.current = params;
+  }, [params]);
```

Comment:
I don't have a preference either way, but I'm curious if you considered wrapping the various `handleX` functions in `useCallback` hooks which could probably achieve the same thing? The outcome would be the same, I'm just curious about your thought process in choosing a ref instead (or maybe it's not that deep 🤷)

Reply:
On second thought, this probably wouldn't solve the closure reference part of the issue so nvm

Reply:
Yeah, I had the same line of thinking, and it's what I tried at first. I was dismayed that it didn't make any difference. I still don't understand how in this case some variable other than the current value of the state variable is getting referenced from closures. I pestered Copilot for a solution and this was the one it suggested that finally worked!